.github/workflows/update.yaml

```
@@ -1,26 +1,82 @@
-name: Update Lean
+name: Update and deploy
+# Everyday attempt to update Lean and mathlib. If the update succeeds then deploy the updated version to the webpage.
+
 on:
   schedule:
-    - cron: "0 11 * * 4"    # every Wednesday at 11:00 UTC
-  push:
-    branches:
-      - main
-  workflow_dispatch:
+    - cron: "0 8 * * *" # every day at 8:00 UTC
+  workflow_dispatch: # allows workflow to be triggered manually
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  update_lean:
+  update:
```

Run on main branch only.

```suggestion
  update:
    if: github.ref == 'refs/heads/main'
```